jobs.go

```diff
@@ -76,16 +76,19 @@ type Job struct {
 	FailureCount       int64          `json:"failure_count"`
 	FailureThreshold   int64          `json:"failure_threshold"`
 	FileMetadata       []FileMetadata `json:"file-metadata"`
-	FilterFiles        []string       `json:"filter_files"` //comes from config, not upstream service
-	Group              string         `json:"group"`        //untested for now
+	FilterFiles        []string       `json:"filter_files"`       //comes from config, not upstream service
+	Group              string         `json:"group"`              //untested for now
+	InputTicketsFile   string         `json:"inputs_ticket_list"` //path to a list of inputs with tickets (not from upstream).
 	InvocationID       string         `json:"uuid"`
 	IRODSBase          string         `json:"irods_base"`
 	Name               string         `json:"name"`
 	NFSBase            string         `json:"nfs_base"`
 	Notify             bool           `json:"notify"`
 	NowDate            string         `json:"now_date"`
-	OutputDir          string         `json:"output_dir"`   //the value parsed out of the JSON. Use OutputDirectory() instead.
-	RequestDisk        string         `json:"request_disk"` //untested for now
+	OutputDir          string         `json:"output_dir"`         //the value parsed out of the JSON. Use OutputDirectory() instead.
+	OutputDirTicket    string         `json:"output_dir_ticket"`  //the write ticket for output_dir (assumes output_dir is set correctly).
+	OutputTicketFile   string         `json:"output_ticket_list"` //path to the file of the output dest with ticket (not from upstream).
```

Comment:
What are these `File` ones and what does "not from upstream" mean?

Reply:
And why are their JSON keys `_list`, I guess?

Reply:
These will be populated by `condor-launcher`, so they don't come from the JSON submitted by the `apps` service (I was being lazy and copied this terminology from other comments in this Job JSON).

`condor-launcher` will create these "Path List" files, but only if input/output tickets exist, then populate the fields with a path to the files generated on the submit node. Then if these `_list` fields are not blank, the paths to these files will be added to the `transfer_input_files` list in `iplant.cmd`.

Reply:
Ah, cool, makes sense. Thanks for the explanation!